tools/tsp-client/src/fs.ts

```diff
@@ -0,0 +1,89 @@
+  await mkdir(path, { recursive: true });
+}
+
+export async function removeDirectory(path: string) {
+  await rm(path, { recursive: true, force: true });
+}
+
+export async function createTempDirectory(outputDir: string): Promise<string> {
+  const tempRoot = path.join(outputDir, "TempTypeSpecFiles");
+  await mkdir(tempRoot, { recursive: true });
+  Logger.debug(`Created temporary working directory ${tempRoot}`);
+  return tempRoot;
+}
+
+export async function writeFileTree(rootDir: string, files: FileTreeResult["files"]) {
+  for (const [relativeFilePath, contents] of files) {
+    const filePath = path.join(rootDir, relativeFilePath);
+    await ensureDirectory(path.dirname(filePath));
+    Logger.debug(`writing ${filePath}`);
+    await writeFile(filePath, contents);
+  }
+}
+
+export async function tryReadTspLocation(rootDir: string): Promise<string | undefined> {
+  try {
+    const yamlPath = path.resolve(rootDir, "tsp-location.yaml");
+    const fileStat = await stat(yamlPath);
+    if (fileStat.isFile()) {
+      const fileContents = await readFile(yamlPath, "utf8");
+      const locationYaml = parseYaml(fileContents);
+      const { directory, commit, repo } = locationYaml;
+      if (!directory || !commit || !repo) {
+        throw new Error("Invalid tsp-location.yaml");
+      }
+      // make GitHub URL
+      return `https://raw.githubusercontent.com/${repo}/${commit}/${directory}/`;
+    }
+  } catch (e) {
+    Logger.error(`Error reading tsp-location.yaml: ${e}`);
+  }
+  return undefined;
+}
+
+export async function readTspLocation(rootDir: string): Promise<[string, string, string, string[]]> {
+  try {
+    const yamlPath = path.resolve(rootDir, "tsp-location.yaml");
+    const fileStat = await stat(yamlPath);
+    if (fileStat.isFile()) {
+      const fileContents = await readFile(yamlPath, "utf8");
+      const locationYaml = parseYaml(fileContents);
+      let { directory, commit, repo, additionalDirectories } = locationYaml;
+      if (!directory || !commit || !repo) {
+        throw new Error("Invalid tsp-location.yaml");
+      }
+      Logger.info(`Additional directories: ${additionalDirectories}`)
+      if (!additionalDirectories) {
+        additionalDirectories = [];
+      }
+      return [ directory, commit, repo, additionalDirectories ];
+    }
+  } catch (e) {
+    Logger.error(`Error reading tsp-location.yaml: ${e}`);
+  }
+  throw new Error("Could not find tsp-location.yaml");
+}
+
+
+export async function getEmitterFromRepoConfig(emitterPath: string): Promise<string | undefined> {
+  await access(emitterPath);
+  const data = await readFile(emitterPath, 'utf8');
+  const obj = JSON.parse(data);
+  if (!obj || !obj.dependencies) {
+    throw new Error("Invalid emitter-package.json");
+  }
+  const languages: string[] = ["@azure-tools/typespec-", "@typespec/openapi3"];
+  for (const lang of languages) {
+    const emitter = Object.keys(obj.dependencies).find((dep: string) => dep.startsWith(lang));
```

Comment:
`find` will always return a matching item or `undefined`, which means the `return` two lines down will always happen and we'll never look for `@typespec/openapi3` or throw the Error at the bottom.

I think we need to test if `emitter` is truthy before returning? And perhaps return undefined at the bottom instead of throwing?

Reply:
Done, still throwing at the bottom because we need an emitter package to be able to generate.